### PR TITLE
TVB-2719 gdist.pyx: New API distance_matrix_of_selected_points

### LINKED ISSUE #52: Add API call for pairwise distance computation
Related to #51.

A new API for this:

> Let's say that I have 180 points. On the first iteration I take the first one and compute the distance to the other 179, on the following iteration i take the second one and compute the distance to the next 178, and so on.

In particular, for `n` points we would consider the 1st one as source and rest of the `(n - 1)` ones as target and, then 2nd one as source and `(n - 2)` as target and so on...

### PR DESCRIPTION
**Pull request by ayan-b**
Closes #52 

See also https://github.com/the-virtual-brain/tvb-gdist/issues/51#issuecomment-663522847.

This is on top of #53.

### CODE REVIEW

**Comment by liadomide** [comment]
as this one is on top of #53 it will be best if you rebase it to trunk before we merge.

File: tests/test_gdist.py [hunk 1]
```diff
@@ -105,3 +105,21 @@ def test_hedgehog_mesh(self):
...
+        vertices = data[0:121].astype(np.float64)
+        triangles = data[121:].astype(np.int32)
+        points = np.array([2, 5, 10], dtype=np.int32)
+        distances = gdist.distance_matrix_of_selected_points(
+            vertices,
+            triangles,
+            points,
+        )
+        epsilon = 1e-6
+        expected = np.loadtxt("data/flat_triangular_mesh_pairwise_matrix.txt")
```
Comment (liadomide) [inline tests/test_gdist.py:157]: Could this unit-test be improved a bit:
- assert that the result is a sparse matrix
- assert that the number of rows/columns not empty related to the `points` size
?

**Review by liadomide — CHANGES_REQUESTED** [review]

**Comment by liadomide** [comment]
And one more thing: pls update also the README documenting the top API calls, now that we have a new entry.
Version can stay 2.1, as we haven't yet released anything since 2.0.2

File: tests/test_gdist.py [hunk 2]
```diff
@@ -138,3 +140,25 @@ def test_hedgehog_mesh(self):
...
+        np.testing.assert_array_almost_equal(distances.toarray(), expected)
+        # test if the obtained matrix is symmetric
+        assert (abs(distances - distances.T) > epsilon).nnz == 0
+        # make sure number of non-zero elements are twice of number of points
+        assert distances.nnz == 2 * points.shape[0]
```
Comment (liadomide) [inline tests/test_gdist.py]: Does this assert  `== 2 * N` scale when you use a different no of points ?
Shouldn't it be `N ^ 2 - N` ? 
`- N ` for the diagonal
Where `N == points.shape[0]`

Comment (ayan-b) [reply tests/test_gdist.py]: Yes, you are right. Thanks!

Comment (ayan-b) [reply tests/test_gdist.py]: Fixed it.

**Review by liadomide — APPROVED** [review]

**Pull request merged**